### PR TITLE
Mup scaling

### PR DESCRIPTION
**Pull request by mehraakash**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog

- Add specific line by line info of high level changes in this PR.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

## Summary by CodeRabbit

* **New Features**
  * Added μP (Maximal Update Parameterization) configuration support to the optimizer, enabling per-parameter-class learning rate overrides for improved training stability.

* **Tests**
  * Added comprehensive unit and functional tests validating μP configuration, override application, and end-to-end training workflows with μP enabled.

### CODE REVIEW

File: src/megatron/bridge/training/optim.py [hunk 1]
```diff
@@ -59,6 +64,22 @@ def setup_optimizer(
         OptimizerConfigOverrideProviderContext(scheduler_config, optimizer_config, model)
     )
 
+    # Apply μP optimizer scaling if enabled on the model config
```
Comment (yaoyu-33) [inline src/megatron/bridge/training/optim.py:68]: is it possible to do this in config container. validate()? all the overriding logic better put in the same place. If not possible, okay to leave it here.

Comment (mehraakash) [reply src/megatron/bridge/training/optim.py:68]: Good point. I think this can't live in `ConfigContainer.validate()` because the μP optimizer overrides require the **post-DDP-wrapped model** — specifically, `get_model_config()` needs to unwrap the DDP/FSDP shell to reach the underlying `TransformerConfig`. At `validate()` time the model hasn't been constructed yet (the config container is built before `setup_model()` is called). By the time `setup_optimizer()` is called, the model is fully wrapped and `get_model_config()` can safely retrieve `use_mup` and `mup_width_mult`.
